### PR TITLE
Refactor use of `plotting_options`

### LINKED ISSUE #12: Refactor use of `plotting_options`
## Description
Refactor use of `plotting_options` in the different graphs.

Instead of many lines `if` statement, dynamically call method through iteration in a `dict`.

## Additional information
This issue follow the work done in https://github.com/motleystate/moonstone/pull/9/files

### PR DESCRIPTION
**Pull request by AgnesBaud**
#### Changelogs

* Refactor use of plotting_options : iteration through dict.keys() + exported in `BaseGraph` (Parent Class)
* `plotting_options`, `output_file` and show dealt with in `plot_one_graph` (instead of at the instantiation)

#### Issue(s)

Fixes #12 

## Definition of Done

* [x]  New code is tested with unit tests
* [x]  Documentation has been updated
* [x]  Pull Request has been revised by a maintainer of the project

### CODE REVIEW

File: tests/utils/test_plot.py [hunk 1]
```diff
@@ -17,17 +17,28 @@ def test_check_list_type(self):
...
+                             {'traces': {'marker_color': 'red'}})
+
+    def test_add_default_titles_to_plotting_options(self):
+        title="graph about something"
+        xlabel="something"
+        ylabel="samples"
+
+        # test with empty dictionary
+        plotting_options = {}
+        print(add_default_titles_to_plotting_options(plotting_options, title, xlabel, ylabel))
```
Comment (khillion) [inline tests/utils/test_plot.py]: I think you forgot to remove a print statement here and replace it by the equality test between dictionnaries

**Review by khillion — COMMENTED** [review]
Looks good to me, we have a good base now let's see in practice how smooth it is to use it.

**Review by khillion — APPROVED** [review]

File: tests/utils/test_plot.py (hunk 1 above)
Comment (AgnesBaud) [reply tests/utils/test_plot.py]: Yep, I saw it when I was correcting for flake8 ! :)

**Review by khillion — APPROVED** [review]
Good to me!

**Pull request merged**